### PR TITLE
LINUXCN-16 Filter the Next Boot dropdown to only show vaild platform images

### PR DESCRIPTION
**Pull request by travispaul**
On CNs that have been setup, only show PIs for the same OS. On CNs not yet setup, show all PIs and make it clear which OS is being selected.

The screenshots below show a headnode with the following PIs installed:

```
# sdcadm platform list -s os
VERSION           CURRENT_PLATFORM  BOOT_PLATFORM  LATEST  DEFAULT  OS
20221107T154729Z  0                 0              false   false    linux
20230227T175258Z  0                 0              false   false    linux
20230530T183117Z  1                 1              true    false    linux
20230406T001127Z  1                 1              false   false    smartos
20230504T000449Z  0                 0              false   false    smartos
20230518T015119Z  0                 0              false   true     smartos
20230601T000726Z  1                 1              true    false    smartos
```

#### Before

All PIs were listed with no way to differentiate Linux from SmartOS:

image

#### After

Now only showing Linux PIs:

Screenshot 2023-06-28 at 11 47 26 AM

#### Before

All PIs were listed with no way to differentiate SmartOS from Linux:

image

#### After

Now only showing SmartOS PIs:

Screenshot 2023-06-28 at 11 48 41 AM

#### Before

All PIs were listed with no way to differentiate SmartOS from Linux:

image

#### After

Now showing all PIs grouped by OS:

Screenshot 2023-06-28 at 12 05 27 PM

### On the default boot option dropdown show all PIs grouped by OS

This could be presented a bit nicer but this dropdown is constructed differently from the one on the Servers page so I opted for the smallest change that provided the most value instead of a larger refactor.

#### Before

All PIs were listed with no way to differentiate SmartOS from Linux:

image

#### After

OS is shown along with each PI version sorted by OS and PI version.

Screenshot 2023-06-28 at 12 06 39 PM
 

### New Favicon

image

### Operating System is shown on the Platform table of the Server view

Screenshot 2023-06-09 at 11 23 16 AM

Screenshot 2023-06-09 at 11 23 08 AM

## Testing Notes

Screenshots were captured from Admin UI running in a local Triton with these changes. I also changed the PI using the new drop down and confirmed the selected PI was still shown on page refresh and that a reboot of the CN resulted in the correct PI.
In the case of the default boot PI, I also confirmed it was set as default in `sdcadm platform list` and that a new CN booted into the selected version.

### CODE REVIEW

**Comment by travispaul** [comment]
Also installed locally via `sdcadm update -C experimental adminui@PR-268-20230606T202130Z-gb66afc4`

File: www/js/models/platforms.js [hunk 1]
```diff
@@ -6,31 +6,58 @@
...
 
 var Backbone = require('backbone');
-var _ = require('underscore');
+
+// Sort by os, then platform image version
+function sortPlatforms (a, b) {
+    if (a.os === b.os) {
+        return a.version > b.version ? -1 : 1;
+    }
+    return a.os > b.os ? -1 : 1;
```
Comment (bahamat) [inline www/js/models/platforms.js:19]: Does this sort order ensure that SmartOS platforms are listed first for unsetup CNs? (that seems to be the case...)

File: www/js/models/platforms.js [hunk 2]
```diff
@@ -6,31 +6,58 @@
...
+                os: platform.os
             });
-            if (n.latest && forServer) {
-                arr.push({
-                    version: 'latest',
-                    latest: n.latest,
-                    label: 'latest (' + d + ')'
+
+            // If this platform is the latest, push an alias for version "latest"
+            if (platform.latest) {
```
Comment (bahamat) [inline www/js/models/platforms.js:49]: Do we still need `&& forServer` in this `if` clause?

**Review by bahamat — CHANGES_REQUESTED** [review]
In addition to other changes, I'd like to have the OS (or system_type) added to the Platform table.

File: lib/servers.js [hunk 3]
```diff
@@ -41,6 +42,18 @@ function getSetupStatus(req, server, callback) {
...
+ * The `system_type` attribute on servers (from sysinfo) differs from
+ * the `os`` property returned from the /platform CNAPI endpoint. Add a
+ * property that matches what `/platform?os=true` returns.
+ */
+function getPlatformOS(systemType) {
+    var systemTypes = {
+        SunOS: 'smartos',
+        Linux: 'linux'
+    };
+    return systemTypes[systemType] || 'unknown';
```
[the remark this answers is not in the document]
Comment (travispaul) [reply lib/servers.js]: Makes sense, I made the default `smartos`.

Let me know if I need to reword the comment, my understanding was that `system_type` ultimately comes from `sysinfo["System Type"]` IIRC I traced it back to here but maybe the comment is still accurate: https://github.com/TritonDataCenter/sdc-cnapi/blob/d6e7bb85a2126fd73ffceeb12db57888e5213d2f/lib/models/server.js#L88

File: www/js/models/platforms.js (hunk 1 above)
Comment (travispaul) [reply www/js/models/platforms.js:19]: Yes, I made sure that was the case. That will be the behavior on both unsetup CNs and on the _Default Server Boot Options_ dropdown.

File: www/js/tpl/server.hbs [hunk 4]
```diff
@@ -6,6 +6,7 @@
 
 <!--
     Copyright 2019 Joyent, Inc.
+    Copyright 2023 MNX Cloud, Inc.
```
[the remark this answers is not in the document]
Comment (travispaul) [reply www/js/tpl/server.hbs:9]: I had some changes I reverted but now that we've put the OS on Platform table it's relevant again.

File: www/js/models/platforms.js (hunk 2 above)
Comment (bahamat) [reply www/js/models/platforms.js:49]: Yeah, I think I'd prefer for latest to be shown only once. I think it's probably a bug that it shows twice.

**Review by bahamat — APPROVED** [review]

**Pull request merged**